provisioning/image_cache.go

```diff
@@ -77,9 +76,32 @@ func imageCacheConfig(targetNamespace string, config metal3iov1alpha1.Provisioni
 		Scheme: "http",
 		Host: net.JoinHostPort(fmt.Sprintf("%s.%s.svc.cluster.local", stateService, targetNamespace),
 			baremetalHttpPort),
-		Path: fmt.Sprintf("/images/%s/%s", imageName, imageName),
+		Path: fmt.Sprintf("/images/%s/cached-%s", imageName, imageName),
+	}
+	return cacheURL.String(), nil
+}
+
+func imageCacheConfig(targetNamespace string, config metal3iov1alpha1.ProvisioningSpec) (*metal3iov1alpha1.ProvisioningSpec, error) {
+	// If the URL contains a comma-seperated list and are live images, split and transform the URLs
+	if strings.Contains(config.ProvisioningOSDownloadURL, "-live") && strings.Contains(config.ProvisioningOSDownloadURL, ",") {
+		var allURLs string
+		URLs := strings.Split(config.ProvisioningOSDownloadURL, ",")
+		for _, URL := range URLs {
+			newURL, err := transformURL(targetNamespace, URL)
+			if err != nil {
+				return nil, err
+			}
+			allURLs = allURLs + newURL + ","
+		}
+		config.ProvisioningOSDownloadURL = strings.TrimSuffix(allURLs, ",")
```

Comment:
nit: you could add them to a slice and then use strings.Join() instead.

Reply:
TIL. Thanks. I'll make the change.